examples/voice_agents/sms_agent.py

```diff
@@ -0,0 +1,98 @@
+import logging
+
+from dotenv import load_dotenv
+
+from livekit.agents import (
+    Agent,
+    AgentServer,
+    AgentSession,
+    JobContext,
+    RunContext,
+    TextMessageContext,
+    cli,
+)
+from livekit.agents.llm import function_tool
+from livekit.plugins import silero
+from livekit.plugins.turn_detector.multilingual import MultilingualModel
+
+# uncomment to enable Krisp background voice/noise cancellation
+# from livekit.plugins import noise_cancellation
+
+logger = logging.getLogger("basic-agent")
+
+load_dotenv()
+
+
+class MyAgent(Agent):
+    def __init__(self, *, greet_on_enter: bool = True) -> None:
+        super().__init__(
+            instructions="Your name is Kelly. You would interact with users via voice."
+            "with that in mind keep your responses concise and to the point."
+            "do not use emojis, asterisks, markdown, or other special characters in your responses."
+            "You are curious and friendly, and have a sense of humor."
+            "you will speak english to the user",
+        )
+        self._greet_on_enter = greet_on_enter
+
+    async def on_enter(self):
+        if self._greet_on_enter:
+            logger.debug("greeting the user")
+            self.session.generate_reply(allow_interruptions=False)
+
+    # all functions annotated with @function_tool will be passed to the LLM when this
+    # agent is active
+    @function_tool
+    async def lookup_weather(
+        self, context: RunContext, location: str, latitude: str, longitude: str
+    ):
+        """Called when the user asks for weather related information.
+        Ensure the user's location (city or region) is provided.
+        When given a location, please estimate the latitude and longitude of the location and
+        do not ask the user for them.
+
+        Args:
+            location: The location they are asking for
+            latitude: The latitude of the location, do not ask user for it
+            longitude: The longitude of the location, do not ask user for it
+        """
+
+        logger.info(f"Looking up weather for {location}")
+
+        return "sunny with a temperature of 70 degrees."
+
+
+server = AgentServer()
+
+
+@server.sms_handler()
```

Comment:
let's position this as a text_handler, instead of sms. getting text right would mean this is accessible on LK Cloud: `https://myapp.livekit.cloud/agents/name/handler`